tests/ssl-opt.sh

```diff
@@ -5561,6 +5566,7 @@ run_test    "Authentication: client no cert, server optional" \
 
 requires_openssl_tls1_3
 requires_key_exchange_with_cert_in_tls12_or_tls13_enabled
+requires_config_enabled MBEDTLS_ECDH_C
```

Comment:
Using `ECDH_C` is what's causing the testing disparities reported by the CI: when `USE_PSA` is enabled (which is the case for the driver-only reference/accelerated components, we want to be testing for `PSA_WANT_ALG_ECDH` instead.

I think we should extend the existing function `requires_pk_alg` to accept "ECDH" as an argument and handle it in a similar way to "ECDSA", then use `requires_pk_alg "ECDH"` here and in other places that need ECDH.

Note: I find that our dependencies in this script are getting increasingly complex and hard to navigate. I'll think if there is any way we can make the situation better. Finding the right way to express dependencies should not require that much effort.

Reply:
I [replaced ](https://github.com/Mbed-TLS/mbedtls/pull/7627/commits/8ff37d41559144f3e1190f8dc38d0063a246d291)`ECDH_C` dependencies as described in above [comment](https://github.com/Mbed-TLS/mbedtls/pull/7627#issuecomment-1609008941). The ssl-opt tests are passing now for `component_test_tls13_only_ephemeral_ffdh` and `component_test_tls13_only_psk_ephemeral_ffdh` configurations.

Reply:
CI is green now :-)